helix-core/src/syntax.rs

```diff
@@ -639,6 +639,8 @@ impl Syntax {
         // Convert the changeset into tree sitter edits.
         let edits = generate_edits(old_source, changeset);
 
+        let mut modified = HashMap::new();
```

Comment:
Might be more efficient to use a `Vec` with `contains`? We don't expect this to contain a lot of elements.

If not, would it make sense to use a `HashSet` instead of a `HashMap`?

Reply:
This will often contain lots of elements. For example if text is inserted at the start of the document, every injection is moved back so there is an entry for every element.

A `HashSet` is not sufficient there are three possible states:
* Injection does not need to be updated (no entry)
*  Injection needs to be moved (call tree.edit: false)
* Injection needs to be reparsed (call tree.edit, parse: true)


I switched to a SecondaryMap from slot_map (basically a dense map) but I will check if there is a more efficient approach

Reply:
I think it would be more efficient to store this state (and the `touched` status) directly within the layer as a bitset. The memory overhead of a single integer per layer is hardly relevant and we don't need to allocate/deallocate and index any additional collections